hack/generated/pkg/reflecthelpers/reflect_helpers.go

```diff
@@ -208,3 +181,75 @@ func FindResourceReferences(transformer genruntime.ARMTransformer) (map[genrunti
 
 	return result, nil
 }
+
+type resolved struct {
+	ResolvedDetails   genruntime.ConvertToARMResolvedDetails
+	ResourceHierarchy genruntime.ResourceHierarchy
+}
+
+func resolve(
+	ctx context.Context,
+	resolver *genruntime.Resolver,
+	metaObject genruntime.MetaObject) (*resolved, error) {
+
+	resourceHierarchy, err := resolver.ResolveResourceHierarchy(ctx, metaObject)
+	if err != nil {
+		return nil, err
+	}
+
+	// Find all of the references
+	refs, err := FindResourceReferences(metaObject)
+	if err != nil {
+		return nil, errors.Wrapf(err, "finding references on %q", metaObject.GetName())
+	}
+
+	// resolve them
+	resolvedRefs, err := resolver.ResolveReferencesToARMIDs(ctx, refs)
```

Comment:
Minor: `resolver.ResolveReferencesToARMIDs` should return a `genruntime.ResolvedReferences`.

Reply:
Yeah you're right - fixed.